### PR TITLE
Add warn if kvm driver version is old

### PR DESCRIPTION
**Pull request by josedonizetti**
issue: https://github.com/kubernetes/minikube/issues/4658

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -850,3 +853,52 @@ func saveConfig(clusterConfig cfg.Config) error {
...
+		if err != nil {
+			console.Warning("Error checking driver version: %v", err)
+			return
+		}
+
+		v := extractVMDriverVersion(string(output))
+
+		// if the driver doesn't have return any version, it is really old, we force a upgrade.
+		if len(v) == 0 {
+			exit.WithCode(exit.Failure, "Please upgrade the 'docker-machine-driver-kvm2'.")
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go]: lets include a link to the docs or where they can download the kvm2 driver

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -850,3 +853,52 @@ func saveConfig(clusterConfig cfg.Config) error {
...
+		}
+
+		minikubeVersion, err := version.GetSemverVersion()
+		if err != nil {
+			console.Warning("Error parsing minukube version: %v", err)
+			return
+		}
+
+		if vmDriverVersion.LT(minikubeVersion) {
+			console.Warning("The 'docker-machine-driver-kvm2' version is old. Please consider upgrading.")
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go]: same here, we could link to a page where they can easily find the newest kvm2.

( i noticed our docs page is a bit confusion, as part of this PR we could make that doc page for kvm2 super simple.( we have too many ifs in the doc)

File: cmd/minikube/cmd/start.go [hunk 3]
```diff
@@ -850,3 +853,52 @@ func saveConfig(clusterConfig cfg.Config) error {
...
+		}
+	}
+}
+
+func extractVMDriverVersion(s string) string {
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go]: please add a comment with example output so it is clear for the next person what this func is doing

File: cmd/minikube/cmd/start_test.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
```
Comment (medyagh) [inline cmd/minikube/cmd/start_test.go]: 2019

**Comment by josedonizetti** [comment]
@medyagh Let me know what you think about the changes. I've added a constant to the kvm documentation, we could start enforcing it everywhere we link to the documentation, that way if we need to change one day, it's a single change. 

I'll try to simplify the documentation now. Although I'm intrigued the PR https://github.com/kubernetes/minikube/pull/4681 make this one unnecessary.

**Comment by medyagh** [comment]
I think this PR could be used to detect the too old KVM drivers that we want to enforce the upgrade ( before the user waits a few minutes till the error happen) if we know the kvm driver is too old. we should end gracefuly without error.

the other PR is good when minikube crashes with an error. it is meant as an after-happened handling, so we have less support ticket.

so this is PR is still helpful and needed to detect sooner

**Comment by josedonizetti** [comment]
@medyagh I've added an upgrade section to the KVM documentation. I know you mentioned we had too many `ifs`, but it kind of seems necessary to explain about dependencies, and possible problems with libvirtd. Wdyt?

**Comment by medyagh** [comment]
@minikube-bot OK to test

**Pull request merged**